src/commands/list.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies.
+ */
+const tableImport = require( 'table' );
+const { table } = tableImport;
+const chalk = require( 'chalk' );
+const logSymbols = require( 'log-symbols' );
+
+/**
+ * Internal dependencies.
+ */
+const { execSync } = require( 'child_process' );
+const envUtils = require( '../env-utils' );
+const makeCommand = require( '../utils/make-command' );
+
+// Add command name, alias and description.
+exports.aliases = [ 'ls' ];
+exports.command = 'list';
+exports.desc = 'Lists all the environments and meta information.';
+
+exports.handler = makeCommand( chalk, logSymbols, async () => {
+    // Get all the environments and initialize a status array.
+    const environments = await envUtils.getAllEnvironments();
+    const envStatus = [ [ 'Name', 'Status', 'URL' ] ];
+
+    // Loop through each environment and add details.
+    for ( const envSlug of environments ) {
+        // Get path of the current environment to perform dedicated checks.
+        const envPath = await envUtils.envPath( envSlug );
+
+        // Get current environment host name, use the starting index.
+        const envHosts = await envUtils.getEnvHosts( envPath );
+        const hostName = envHosts[0];
+
+        try {
+            // Get the container list for current environment passing name of the slug to filter.
+            const output = execSync( `docker ps --filter "name=${ envSlug }"` ).toString();
```

Comment:
Use [dockerode](https://www.npmjs.com/package/dockerode) to call docker API to retrive containers. You can load a docker API client using `makeDocker` function:

```js
const makeDocker = require( '../utils/make-docker' );

...

const docker = makeDocker();
const containers = await docker.listContainers( options );
```